tests/common/plugins/psu_controller/snmp_psu_controllers.py

```diff
@@ -51,24 +101,35 @@ def _get_pdu_ports(self):
         The PDU ports connected to DUT must have hostname of DUT configured in port name/description.
         This method depends on this configuration to find out the PDU ports connected to PSUs of specific DUT.
         """
-        try:
-            cmd = "snmpwalk -v 1 -c public -Ofenq %s %s " % (self.controller, self.PORT_NAME_BASE_OID)
-            stdout = run_local_cmd(cmd)
-            for line in stdout.splitlines():
-                if self.hostname in line:  # PDU port name/description should have DUT hostname
-                    fields = line.split()
-                    if len(fields) == 2:
+        if not has_pysnmp:
```

Comment:
"The PDU ports connected to DUT must have hostname of DUT configured in port name/description.
        This method depends on this configuration to find out the PDU ports connected to PSUs of specific DUT."

about this assumption, i do not think it is a good assumption. 

I think the we should define the pdu ports in the https://github.com/Azure/sonic-mgmt/blob/master/ansible/files/lab_connection_graph.xml

we should not have external dependency on on this pdu description. 

if we have that assumption, then there is extra work to configure the description on those pdu ports, who is going to do that.

Reply:
@wangxin , can you comment on this?

Reply:
I agree. Enforcing hostname configured in PDU port description is a temporary workaround. The correct solution is to store the lab connection information in configuration files and read from them during testing. To support that, I think the conn_graph_facts module may need some enhancement. Currently all the logic of conn_graph_facts is about network interface connections.

Reply:
ok. I think this is bigger change. Let's do this in future PR. @sujinmkang , we need some discussion with @yxieca to close the design part.

Reply:
I agree that PDU mapping should be defined and maintained.

But I not sure it should be included in the lab_connection_graph.xml. It would make PDU looking like 'roots' in the graph.

Not sure if our existing script can handle it properly? We should definitely take this into future design/PRs.